nemo/core/config/set_config.py

```diff
@@ -24,15 +24,22 @@
 
 
 def hydra_runner(
-    config_path: Optional[str] = None, config_name: Optional[str] = None
+    config_path: Optional[str] = None, config_name: Optional[str] = None, schema: Optional[Any] = None
 ) -> Callable[[TaskFunction], Any]:
     """
     Decorator used for passing the Config paths to main function.
+    Optionally registers a schema used for validation/providing default values.
 
     Args:
-        config_path: path to the directory where the config exists.
-        config_name: name of the config file.
+        config_path: Path to the directory where the config exists.
+        config_name: Name of the config file.
+        schema: Structured config  type representing the schema used for validation/providing default values.
     """
+    if schema is not None:
+        # Create config store.
+        cs = ConfigStore.instance()
+        # Register the configuration as a node under a given name.
+        cs.store(name=config_name, node=schema)
```

Comment:
While I like this as a first step, this seems extremely limited unless we have a way of composing schemas. It seems redundant to have to copy and paste the `TrainerConfig` along with every new schema we create.

I would like to see a more complete solution that allows us to just drop in a model schema.

Reply:
Not sure if I understand the copy and paste part. The `TrainerConfig` is implemented once and then used as part of other configs...

Note @okuchaiev asked me to scope this PR to only the hydra_runner, there are no examples. But I have several working examples in `main-vis-res` branch, e.g.:
https://github.com/NVIDIA/NeMo/blob/0c5573fd9ac4dff4797d1823e7ab36514164291b/examples/vis/clevr_qtype_training.py#L43

Reply:
@blisc I don't understand part about copy and paste the TrainerConfig. Can't we just import it?

Reply:
@blisc I'll be proposing a ModelPT level structured config in a few days, and an example config dict for ASR as an example. This trainer config can be easily attached I believe to the above.

Reply:
Not sure if I understand... you want to attach trainer config to model config?

Reply:
I guess we should take this off-line (as it is out of scope of this PR), but model and trainer are separate sections -> separate Structured Configs.

In my solutions it is AppConfig that "ties" them together, e.g. like in here:
https://github.com/NVIDIA/NeMo/blob/0c5573fd9ac4dff4797d1823e7ab36514164291b/examples/vis/clevr_qtype_training.py#L29
```
@dataclass
class AppConfig:
    """
    This is structured config for this application.
    Args:
        trainer: configuration of the trainer.
        model: configuation of the model.
    """

    model: QTypeConfig = QTypeConfig()
    # ...
    trainer: TrainerConfig = TrainerConfig()
```

(removed other sections to avoid even more unrelated discussion ;) )

Reply:
No no. I'll be sending a draft proposed Model PT config class later next week. I'll be using the above Trainer config there.

Reply:
👍

Reply:
just please make all *_ds optional ;)

Reply:
I see, I missed that `AppConfig` was tying them together. I thought the passed `schema` would require everyone to write their own schema that contained `TrainerConfig`